Test test_storage_s3_queue/test_alter_settings is flaky

**Link to report**
https://s3.amazonaws.com/clickhouse-test-reports/0/06b1ef2c2ddd109218c7b3fc75c75d4ebe96227a/integration_tests__tsan__[1_6].html


**Error message**
```
        for node in [node1, node2]:
            check_int_settings(node, int_settings)
            check_string_settings(node, string_settings)
    
            node.restart_clickhouse()
>           assert expected_rows == get_count()
E           assert 1000 == 1273
E            +  where 1273 = <function test_alter_settings.<locals>.get_count at 0x7f287d1a92d0>()
```

--- a/src/Processors/Transforms/buildPushingToViewsChain.cpp
+++ b/src/Processors/Transforms/buildPushingToViewsChain.cpp
@@ -511,7 +511,7 @@ Chain buildPushingToViewsChain(
     auto log = getLogger("buildPushingToViewsChain");
     LOG_TEST(log, "Views: {}", views.size());
 
-    if (no_destination && views.size() == 0)
+    if (no_destination && views.empty())
         LOG_WARNING(log, "No views attached and no_destination = 1");
 
     ViewsDataPtr views_data;

--- a/src/Processors/Transforms/buildPushingToViewsChain.cpp
+++ b/src/Processors/Transforms/buildPushingToViewsChain.cpp
@@ -508,6 +508,12 @@ Chain buildPushingToViewsChain(
     auto table_id = storage->getStorageID();
     auto views = DatabaseCatalog::instance().getDependentViews(table_id);
 
+    auto log = getLogger("buildPushingToViewsChain");
+    LOG_TEST(log, "Views: {}", views.size());
+
+    if (no_destination && views.size() == 0)
+        LOG_WARNING(log, "No views attached and no_destination = 1");
+
     ViewsDataPtr views_data;
     if (!views.empty())
     {
@@ -516,7 +522,6 @@ Chain buildPushingToViewsChain(
     }
 
     std::vector<Chain> chains;
-
     for (const auto & view_id : views)
     {
         try

--- a/src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.cpp
+++ b/src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.cpp
@@ -550,6 +550,10 @@ bool StorageObjectStorageQueue::streamToViews()
 
     while (!shutdown_called && !file_iterator->isFinished())
     {
+        /// FIXME:
+        /// it is possible that MV is dropped just before we start the insert,
+        /// but in this case we would not throw any exception, so
+        /// data will not be inserted anywhere.
         InterpreterInsertQuery interpreter(
             insert,
             queue_context,
@@ -601,12 +605,12 @@ bool StorageObjectStorageQueue::streamToViews()
         }
         catch (...)
         {
-            commit(/* insert_succeeded */false, sources, getCurrentExceptionMessage(true));
+            commit(/* insert_succeeded */false, rows, sources, getCurrentExceptionMessage(true));
             file_iterator->releaseFinishedBuckets();
             throw;
         }
 
-        commit(/* insert_succeeded */true, sources);
+        commit(/* insert_succeeded */true, rows, sources);
         file_iterator->releaseFinishedBuckets();
         total_rows += rows;
     }
@@ -617,6 +621,7 @@ bool StorageObjectStorageQueue::streamToViews()
 
 void StorageObjectStorageQueue::commit(
     bool insert_succeeded,
+    size_t inserted_rows,
     std::vector<std::shared_ptr<ObjectStorageQueueSource>> & sources,
     const std::string & exception_message) const
 {
@@ -650,8 +655,8 @@ void StorageObjectStorageQueue::commit(
         source->finalizeCommit(insert_succeeded, exception_message);
 
     LOG_TRACE(
-        log, "Successfully committed {} requests for {} sources",
-        requests.size(), sources.size());
+        log, "Successfully committed {} requests for {} sources (inserted rows: {}, successful files: {})",
+        requests.size(), sources.size(), inserted_rows, successful_objects.size());
 }
 
 static const std::unordered_set<std::string_view> changeable_settings_unordered_mode

--- a/src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.h
+++ b/src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.h
@@ -132,6 +132,7 @@ private:
     /// Commit processed files to keeper as either successful or unsuccessful.
     void commit(
         bool insert_succeeded,
+        size_t inserted_rows,
         std::vector<std::shared_ptr<ObjectStorageQueueSource>> & sources,
         const std::string & exception_message = {}) const;
 };

--- a/tests/integration/test_storage_s3_queue/test.py
+++ b/tests/integration/test_storage_s3_queue/test.py
@@ -2280,6 +2280,7 @@ def test_alter_settings(started_cluster):
 
     table_name = f"test_alter_settings_{uuid.uuid4().hex[:8]}"
     dst_table_name = f"{table_name}_dst"
+    mv_name = f"{table_name}_mv"
     keeper_path = f"/clickhouse/test_{table_name}"
     files_path = f"{table_name}_data"
     files_to_generate = 1000
@@ -2326,13 +2327,12 @@ def test_alter_settings(started_cluster):
         started_cluster, files_path, files_to_generate, start_ind=0, row_num=1
     )
 
-    create_mv(node1, f"r.{table_name}", dst_table_name)
-    create_mv(node2, f"r.{table_name}", dst_table_name)
+    create_mv(node1, f"r.{table_name}", f"r.{dst_table_name}", mv_name = f"r.{mv_name}")
 
     def get_count():
         return int(
             node1.query(
-                f"SELECT count() FROM clusterAllReplicas(cluster, default.{dst_table_name})"
+                f"SELECT count() FROM clusterAllReplicas(cluster, r.{dst_table_name})"
             )
         )
 

